Realm/RLMArray.mm

```diff
@@ -149,9 +212,15 @@ - (NSUInteger)indexOfObject:(RLMObject *)object {
     return NSNotFound;
 }
 
+- (void)removeAllObjects {
+    changeArray(self, NSKeyValueChangeRemoval, NSMakeRange(0, _backingArray.count), ^{
+        [_backingArray removeAllObjects];
+    });
+}
+
 - (void)deleteObjectsFromRealm {
     for (RLMObject *obj in _backingArray) {
-        RLMDeleteObjectFromRealm(obj, _realm);
+        RLMDeleteObjectFromRealm(obj, obj->_realm);
```

Comment:
The previous behavior was to crash unless the first object in the array was a persisted object, and throw an exception in that case. This changes it to actually work.